composer.json

```diff
@@ -26,21 +26,18 @@
         },
         "sort-packages": true,
         "allow-plugins": {
-            "dealerdirect/phpcodesniffer-composer-installer": true,
-            "composer/package-versions-deprecated": true,
-            "laminas/laminas-dependency-plugin": true
+            "dealerdirect/phpcodesniffer-composer-installer": true
         }
     },
     "require": {
         "php": "~7.4.0 || ~8.0.0 || ~8.1.0",
+        "composer-plugin-api": "^2.0",
         "laminas/laminas-stdlib": "^3.2.1",
-        "psr/container": "^1.0"
+        "psr/container": "^1.1 || ^2.0"
```

Comment:
Our `composer.lock` didn't get updated with this 🤔

Reply:
What does that actually mean?

Reply:
Means that our static analysis, CS, etc. are not checking if we're compatible with 2.x, nor that a `composer update` will bring us to 2.x

Reply:
Yes, we only verify one version. Thats how we do this in all packages which do support multiple majors of any kind?

Unit tests are running on v2, codestyle has nothing to do with dependency versioning so the only thing I'd say could be useful would be static analysis.

But unit tests are definitely executed for v2:
https://github.com/laminas/laminas-servicemanager/runs/7380249774?check_suite_focus=true#step:3:73

What I could do here is to add `.laminas-ci.json` additional checks to actually run psalm on v2.

Reply:
Hmm, I think the nightly lockfile upgrade will bump us to v2 right after the patch is merged, no?

Reply:
I have no idea if renovate will bump the lock file in this case. Do you suggest to actually have v2 within the lockfile and use a downgrade (`composer require psr/container:^1`) to run psalm for v1 instead of bumping to v2 as I do as of https://github.com/laminas/laminas-servicemanager/pull/127/commits/2d36a4ba947f0c5ed25763de9d5c7ea9c38609bd?

Reply:
I've bumped the lockfile to v2 and downgrade to v1 for CI.

Reply:
Awesome, indeed best approach, thanks @boesing!